From 92994f1cfb79017ae22cb379b14d84346e296b00 Mon Sep 17 00:00:00 2001
From: Monika Korada <quic_koramoni@quicinc.com>
Date: Wed, 11 Sep 2024 10:14:10 +0530
Subject: [PATCH] hostapd: Fix NULL driv_priv pointer access

In WDS REP, after STA getting associated to root AP, REP AP
interfaces will be up. While a MLD interface is getting
enabled for the first time, bss and driv_priv will be created
in hostapd.

If the MLD interface is failed to enable for some reason,
we need to deinit bss and driv_priv to avoid NULL pointer
access.

Change-Id: I178575437afd43c3025bf27dea2dcf5d6e51b274
Signed-off-by: Monika Korada <quic_koramoni@quicinc.com>
Signed-off-by: Karthik M <quic_karm@quicinc.com>
---
 src/ap/hostapd.c | 1 +
 1 file changed, 1 insertion(+)

--- a/src/ap/hostapd.c
+++ b/src/ap/hostapd.c
@@ -4082,6 +4082,7 @@ int hostapd_enable_iface(struct hostapd_
 		return -1;
 
 	if (hostapd_setup_interface(hapd_iface)) {
+		hostapd_bss_link_deinit(hapd_iface->bss[0]);
 		hostapd_deinit_driver(hapd_iface->bss[0]->driver,
 				      hapd_iface->bss[0]->drv_priv,
 				      hapd_iface);
